test/unit/chunked-uploads.js

```diff
@@ -197,6 +200,86 @@ if (qqtest.canDownloadFileAsBlob) {
             });
         }
 
+        function testChunkedEveryFailureAndRecovery(done) {
+            assert.expect(6 + (expectedChunks * 17) + (expectedChunks * 6), done);
```

Comment:
I'll admin I wish I hadn't added the ability to track asserts like this - it has become a maintenance mess. I ask you do one of two things here:

Either heavily comment this line to explain the significance of this number, or rewrite this test (possibly breaking it into smaller ones) such that assert counting is no longer needed.